addons/base/views.py

```diff
@@ -697,22 +697,23 @@ def osfstoragefile_update_view_analytics(self, auth, fileversion):
 
 @file_signals.file_viewed.connect
 def osfstoragefile_viewed_update_metrics(self, auth, fileversion):
-    resource = BaseFileNode.objects.get(versions__id=fileversion.id).target
+    file = BaseFileNode.objects.filter(versions__id=fileversion.id).last()
```

Comment:
`.last` is guaranteed to not throw a `MultipleObjectsReturned` error, but isn't guaranteed to deliver the correct `FileNode` object. Would it be possible to include it in the `file_signals.file_viewed.send` call instead?

Reply:
Won't all the versions have the same BaseFIleNode though? I only retrieve the version, not the full node.

Reply:
FileVersions help dedupe information, so one FileVersion can belong to many BaseFileNodes